torch/csrc/api/include/torch/nn/modules/sequential.h

```diff
@@ -347,16 +356,56 @@ class SequentialImpl : public Cloneable<SequentialImpl> {
   /// The base case, when the list of modules is empty.
   void push_back() {}
 
-  // Box the AnyModules to give Sequential reference semantics, like the rest of
-  // the API. Note that this is not required otherwise, this could just be a
+  // Box the AnyModules to give AnySequential reference semantics, like the rest
+  // of the API. Note that this is not required otherwise, this could just be a
   // `vector<AnyModule>`.
   std::vector<AnyModule> modules_;
+
+  friend class SequentialImpl;
```

Comment:
Is there a reason we need to have `SequentialImpl` as friend class?

Reply:
I did this to access ```modules_``` from ```SequentialImpl``` in clone function. I didn't want to make ```modules_``` protected.

Reply:
We can likely remove the need for this once we don't have `clone` function specifically for SequentialImpl.